assignments/arrays.js

```diff
@@ -70,27 +70,55 @@ console.log(`Car 33 is a *car year goes here* *car make goes here* *car model go
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
-console.log();
+let inventoryLength = inventory.length;
+console.log()
+console.log(inventoryLength);
+console.log(("Make: " +inventory[ inventoryLength-1].car_make + " Model: " +inventory[ inventoryLength-1].car_model));
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+
+let inventoryLength1 = inventory.length;
+
+for (i=0; i< inventoryLength1;i++){
+
+ carModels[i] = inventory[i].car_model
+//  console.log(carModels[i])
+ }
+
+console.log(carModels.sort());
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+for (i=0; i< inventory.length;i++) {
+
+ carYears[i] = inventory[i].car_year;
+ 
+}
+
+ console.log(carYears);
+
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+ let oldCars =[];
+ for (i=0; i< inventory.length;i++) {
+   if(parseInt(inventory[i].car_year) < 2000){
+  oldCars[i]=parseInt(inventory[i].car_year);
```

Comment:
This is a situation where we have holes in our array. Because we are not adding every element in `inventory`, but the index, `i` is still incrementing after every iteration of the loop, we have a bunch of `undefined` elements in the `oldCars` array. If you log `oldCars` to the console, you get something like this:
```
[undefined, undefined, undefined, 1983, 1990, 1995, undefined, 1987, 1996, undefined]
```
If you use the `.push()` method, we don't get those `undefined`s since `.push` doesn't rely on the current index in `inventory` to add the element.